api/views.py

```diff
@@ -44,517 +53,157 @@ def api_root(request):
-        url                           string                  url of associated OSF object (e.g. project url)
-        metadata                      object                  additional information about the item
-        date_created                  iso8601 timestamp       date/time when the item was created
-        date_submitted                iso8601 timestamp       date/time when the item was submitted
-        date_accepted                 iso8601 timestamp       date/time when the item was accepted
-        location                      string                  location of the event item
-        start_time                    iso8601 timestamp       date/time when the event item begins
-        end_time                      iso8601 timestamp       date/time when the event item ends
-        category                      string                  item category (e.g. 'talk', 'poster')
-
-    ##Relationships
-
-    ### Created By
-
-    User who created this item.
-
-    ## Actions
-
-    ###Update
-
-            Method:        PUT / PATCH
-            URL:           /api/collections/<collection_id>/items/<item_id> OR
-                           /api/items/<item_id>
-            Query Params:  <none>
-            Body (JSON):   {
-                             "data": {
-                               "type": "items",                  # required
-                               "attributes": {
-                                 "title":       {title},         # required
-                                 "description": {description},   # optional
-                                 "type":        {type},          # required
-                                 "status":      {status},        # required
-                                 "source_id":   {source_id},     # optional
-                                 "url":         {url},           # optional
-                                 "metadata":    {metadata},      # optional
-                                 "location":    {location},      # optional
-                                 "start_time":  {start_time},    # optional
-                                 "end_time":    {end_time},      # optional
-                                 "category":    {category}       # required
-                               }
-                             }
-                           }
-            Success:       200 OK + item representation
+        collection = Collection.objects.get(self.request.data.get("collection"))
+        if not user.has_perm('collection.AddItem', collection):
+            return HttpResponse('Unauthorized', status=401)
+        serializer.save()
 
-    Note: The `title`, `type`, `status` and `category` fields are required for PUT and optional for PATCH requests.
+    def retrieve(self, request, *args, **kwargs):
+        instance = self.get_object()
+        #if request.user.has_perm("view", instance):
```

Comment:
Remove commented code before merge

Reply:
This is getting uncommented in the next pr